crates/hyperqueue/src/server/event/log/write.rs

```diff
@@ -1,66 +1,44 @@
-use crate::server::event::log::canonical_header;
+use crate::server::event::log::HQ_JOURNAL_HEADER;
 use crate::server::event::{bincode_config, Event};
+use crate::HQ_VERSION;
 use bincode::Options;
+use std::fs::{File, OpenOptions};
+use std::io::{BufWriter, Seek, Write};
 use std::path::Path;
-use tokio::fs::{File, OpenOptions};
-use tokio::io::AsyncWriteExt;
 
 /// Streams monitoring events into a file on disk.
 pub struct EventLogWriter {
-    file: File,
-    buffer: Vec<u8>,
+    file: BufWriter<File>,
 }
 
-const BUF_MAX_SIZE: usize = 16 * 1024;
-
 impl EventLogWriter {
-    pub async fn create_or_append(path: &Path) -> anyhow::Result<Self> {
-        let mut file = OpenOptions::new()
-            .create(true)
-            .append(true)
-            .open(path)
-            .await?;
-        let header = bincode_config().serialize(&canonical_header())?;
-        file.write_all(&header).await?;
-        file.flush().await?;
-
-        // Keep buffer capacity larger than max size to avoid reallocation if we overflow
-        // the buffer.
-        let buffer = Vec::with_capacity(BUF_MAX_SIZE * 2);
-        Ok(Self { file, buffer })
+    pub fn create_or_append(path: &Path) -> anyhow::Result<Self> {
+        let mut file = if !path.exists() {
```

Comment:
It was just a temporary version. Not in the final one.
On the other hand, when some other process will modify the journal, it will probably goes horribly wrong as we read the file, potentially truncating it, and then appending to it. If someone modify the file in between it it is quite wild what may happen with the consistency of the file